pkg/descheduler/strategies/lownodeutilization.go

```diff
@@ -191,19 +198,84 @@ func evictPodsFromTargetNodes(client clientset.Interface, evictionPolicyGroupVer
 		glog.V(3).Infof("evicting pods from node %#v with usage: %#v", node.node.Name, node.usage)
 		currentPodsEvicted := nodepodCount[node.node]
 
-		// evict best effort pods
-		evictPods(node.bePods, client, evictionPolicyGroupVersion, targetThresholds, nodeCapacity, node.usage, &totalPods, &totalCpu, &totalMem, &currentPodsEvicted, dryRun, maxPodsToEvict)
-		// evict burstable pods
-		evictPods(node.bPods, client, evictionPolicyGroupVersion, targetThresholds, nodeCapacity, node.usage, &totalPods, &totalCpu, &totalMem, &currentPodsEvicted, dryRun, maxPodsToEvict)
-		// evict guaranteed pods
-		evictPods(node.gPods, client, evictionPolicyGroupVersion, targetThresholds, nodeCapacity, node.usage, &totalPods, &totalCpu, &totalMem, &currentPodsEvicted, dryRun, maxPodsToEvict)
+		// Check if all pods have priority, if yes, go ahead and sort them based on priority.
+		if checkIfAllPodsHavePriority(node.allpods) {
+			glog.V(1).Infof("All pods have priority associated with them. Evicting pods based on priority")
+			// sort the evictable Pods based on priority.
+			sortPodsBasedOnPriority(node.allpods)
```

Comment:
I am thinking if you could find out how to sort based on 2 parameters: priority and qos field, that way, you will have just one sorted pod list, and you would not have to create any priorityToPodMap or classify pods again. For example, lets say there are 5 unsorted pods as follows (convention is like pod-name(priority, qos) just for example) : p1(5,G), p2(3,G), p3(5,be), p4(2,G)  p5(3,burstable), and after sorting based on priority and qos field, the sorted list would look like: p4(2,G), p5(3,burstable), p2(3,G), p3(5,be), p1(5,G), Once we have this sorted list, we could just start evicting from the first one till the point in the sorted list when our target is met. I think it would simplify the logic a lot.

Reply:
@ravisantoshgudimetla I have another idea too which could be a way better from performance point of view. Fetch the list of the priority classes from apiserver. This way we already know the list of priorities. Create same number of buckets(slices) as number of priorities. Traverse the list of pods one and put each pod in the respective bucket (counting sort). The same algorithm (counting sort which is o(n)) I am using in NodeUtilization for guaranteed, best effort and burstable pods.

Reply:
Even though, the 2nd option LGTM in terms of performance, I want to something on the lines of first option.

```go
type podPriorityMap struct {
priority int32
podList []*v1.Pod
}
```

and then we will sort podPriorityMap struct based on priorities. After this, we can evict pods from podPriorityMap.podList based on QoS tiers. This will ensure that there is one sorting and then one linear search while classification.

Reply:
@ravisantoshgudimetla IMO, we dont need podPriorityMap. Just sorting pod list based on priority and QoS is enough.

Reply:
I think, we can do it, if we allow multiple traversals of the list and you wanted this to be in single traversal unless I am missing something.

Reply:
So, I was trying to do the samething but I see that less works on both the keys that we want to sort. I will try to simplify it and see. https://gist.github.com/ravisantoshgudimetla/b3ed1da28c125abf9d43b3ca1b5b8ed3